src/snovault/elasticsearch/index_logger.py

```diff
@@ -0,0 +1,98 @@
+"""
+Indexing Data Log
+
+* Logs uuid progress, timing, and exceptions as they happen.
+"""
+import logging
+import time
+
+
+class IndexLogger(object):
+    '''Basic Logger tailored to our indexing process'''
+    log_name = 'indexing_time'
+    log_path = './src/snovault/elasticsearch'
+
+    def __init__(self, do_log=False, log_name=None, log_path=None):
+        self._do_log = do_log
+        if log_name:
+            self.log_name = log_name
+        if log_path:
+            self.log_path = log_path
+        self._the_log = None
+
+    def _close_handlers(self):
+        '''Close all log handlers'''
+        for handler in self._the_log.handlers:
+            handler.close()
+            self._the_log.removeHandler(handler)
+
+    def _get_log(self):
+        if self._do_log:
+            file_name = "{}-{}.log".format(
+                self.log_name,
+                str(int(time.time() * 10000000)),
+            )
+            file_path = "{}/{}".format(self.log_path, file_name)
+            level = logging.INFO
+            formatter_str = '%(asctime)s %(message)s'
+            log = logging.getLogger(self.log_name)
+            hanlder = logging.FileHandler(file_path)
+            formatter = logging.Formatter(formatter_str)
+            hanlder.setFormatter(formatter)
+            log.addHandler(hanlder)
+            log.setLevel(level)
+            return log
+        return None
+
+    def _reset_log(self):
+        '''
+        Close handlers and Get new log
+
+        * Logger gets logs name so we call twice to clear and get a new one
+        '''
+        if self._do_log:
```

Comment:
this seems super clunky but I guess you have a good reason?

Reply:
eh...It's clunky.  I didn't like it.  I used the flag in the logger class to limit the indexer code changes.  Now that it works maybe I can find a better solution.